.github/workflows/pre-merge.yml

```
@@ -0,0 +1,36 @@
+# Validates Pull Requests by running linting, testing, and type checking
+name: Pre-merge
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches: [main]
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
```

We might want to avoid having this run for draft PRs as they are used to get feedback and not ready for merge requests.
```suggestion
    runs-on: ubuntu-latest
    if: ${{ !github.event.pull_request.draft }}
```